src/coreclr/tools/aot/ILCompiler.ReadyToRun/IL/Stubs/PInvokeILEmitter.cs

```diff
@@ -46,6 +47,14 @@ private void EmitPInvokeCall(PInvokeILCodeStreams ilCodeStreams)
             // if the SetLastError flag is set in DllImport, clear the error code before doing P/Invoke
             if (_importMetadata.Flags.SetLastError)
             {
+                if (_targetMethod is EcmaMethod ecmaMethod
```

Comment:
This would also be better without the ECMA cast, although non-ECMA things show up more rarely in crossgen2 (for now at least).